File: src/main/java/sawyern/cookiebot/models/entity/WeeklyCooldown.java
package sawyern.cookiebot.models.entity;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "WEEKLY_COOLDOWN")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(callSuper = true)
public class WeeklyCooldown extends DbItem {
    @OneToOne
    @JoinColumn(name = "ACCOUNT_ID")
    private Account account;

    @Column
    private LocalDateTime canUseNext;
}

File: src/main/kotlin/sawyern/cookiebot/models/WeeklyCooldown.kt
package sawyern.cookiebot.models

import javax.persistence.*

@Entity
@Table(name = "WEEKLY_COOLDOWNS")
data class WeeklyCooldown(
    @OneToOne
    @JoinColumn(name = "ACCOUNT_ID")
    var account: Account,

    @Column
    var onCooldown: Boolean
): DbItem()